src/viam/sdk/services/motion/client.cpp

```diff
@@ -45,13 +45,19 @@ std::string MotionClient::move_on_map(
     const Name& component_name,
     const Name& slam_name,
     const std::shared_ptr<motion_configuration>& motion_configuration,
+    const std::vector<GeometryConfig>& obstacles,
```

Comment:
Yep! That's correct.

Reply:
Not exactly "list" - more like "dynamic array". Both `std::vector` and `std::list` are dynamically sized and bidirectionally iterable. The key difference is that `std::vector` is guaranteed contiguous, but insertions/deletions can cost reallocation / element-by-element copies, whereas `std::list` is non-contiguous, but insertions/deletions anywhere are cheap since contiguity isn't in play.